bindings/python/RobotInterface.cpp

```diff
@@ -0,0 +1,106 @@
+/**
+ * @file RobotInterface.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <pybind11/eigen.h>
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include <yarp/dev/PolyDriver.h>
+
+#include "BipedalLocomotion/RobotInterface/IRobotControl.h"
+#include "BipedalLocomotion/RobotInterface/YarpHelper.h"
+#include "BipedalLocomotion/RobotInterface/YarpRobotControl.h"
+
+#include "bipedal_locomotion_framework.h"
+
+void BipedalLocomotion::bindings::CreatePolyDriver(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+    using namespace yarp::dev;
+    py::class_<PolyDriver, std::shared_ptr<PolyDriver>>(module, "PolyDriver");
+}
+
+void BipedalLocomotion::bindings::CreatePolyDriverDescriptor(pybind11::module& module)
+{
+    namespace py = ::pybind11;
+    using namespace BipedalLocomotion::RobotInterface;
+    using namespace BipedalLocomotion::ParametersHandler;
+
+    py::class_<PolyDriverDescriptor>(module, "PolyDriverDescriptor")
+        .def(py::init())
+        .def_readwrite("key", &PolyDriverDescriptor::key)
+        .def_readwrite("poly", &PolyDriverDescriptor::poly)
```

Comment:
Is intended here to have both read and write capabilities?

Reply:
I tried to replicate the c++ implementation. There they are members of a struct

Reply:
Ok. Then, it's up to you if you want to allow and it makes sense from python to override the values. Structs in python are typically mapped as `NamedTuple` if they are immutable, or `dataclass` if they are mutable (but they can be made immutable as well, they are more recent that NamedTuples).